src/backend/opencl/kernel/ireduce_first.cl

```diff
@@ -52,7 +52,7 @@ void ireduce_first_kernel(__global T *oData,
     uint out_idx = xid;
 
     if (cond && xid < lim) {
-        out_val = iData[xid];
+        if (!(IS_NAN(iData[xid]))) out_val = iData[xid];
```

Comment:
This condition should wrap both statements in the body of this if statement. Probably a good idea to add a test for this scenario.

Reply:
So this last one - let me try to see if I got what you're trying to say here. Do you mean that both assignments to `out_val` and `out_idx` should be together in the body of an if-statement?

Reply:
:+1: changed this based on our slack conversation. The is_nan check be moved to the outer condition so that we don't assign out_val and out_idx to new values if the current iData element is nan. just waiting for the checks now